src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -828,14 +836,12 @@ class TrainDialogs extends React.Component<Props, ComponentState> {
 
         try {
             const teachWithHistory = await ((this.props.fetchHistoryThunkAsync(this.props.app.appId, trainDialogWithDefinitions, this.props.user.name, this.props.user.id) as any) as Promise<CLM.TeachWithHistory>)
-            const originalId = this.state.currentTrainDialog
-                ? this.state.currentTrainDialog.trainDialogId
-                : null
+
             this.setState({
                 history: teachWithHistory.history,
                 lastAction: teachWithHistory.lastAction,
                 currentTrainDialog: trainDialog,
-                originalTrainDialogId: originalId,
+                originalTrainDialog: this.state.currentTrainDialog || this.state.currentTrainDialog,
```

Comment:
I'm guessing one of these supposed to be `this.state.originalTrainDialog` ? 

Otherwise remove one of them

Reply:
fixed